my_each.rb

```diff
@@ -1,6 +1,143 @@
 module Enumerable
-    def my_each(array)
-      for i in 0...array.length
-        p "gaa"
+  def my_each
+    (0...length).each do |i|
+      yield self[i]
     end
-end
\ No newline at end of file
+    self
+  end
+
+  def my_each_with_index
+    (0...length).each do |i|
+      yield self[i], i
+    end
+    self
+  end
+
+  def my_select
+    new_array = []
+    my_each do |element|
+      new_array.push(element) if yield element
+    end
+    new_array
+  end
+
+  def my_all?
+    result = true
+    my_each do |element|
+      statement = if block_given?
+                    yield element
+                  else
+                    element
+                  end
+
+      unless statement
+        result = false
+        break
+      end
+
+      result = true
+    end
+    result
+  end
+
+  def my_any?
+    result = true
+    my_each do |element|
+      statement = if block_given?
+                    yield element
+                  else
+                    element
+                  end
+
+      if statement
+        result = true
+        break
+      end
+      result = false
+    end
+    result
+  end
+
+  def my_none?
+    !my_any? do |element|
+      yield element
+    end
+  end
+
+  def my_count
+    counter = 0
+    my_each do |element|
+      param = yield element
+      counter += 1 if param.eql?(true) || param.eql?(element)
+    end
+    counter
+  end
```

Comment:
- [ ] should have working `my_count` identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- Your method doesn't consider the possibility when no argument or block is given. It should return the count of all elements inside the enumerable.

- Your method doesn't consider the possibility when a number is given as an argument `(some_enumerable.my_count(5)).` It should return the count of all elements that match the argument.